plone/schema/jsonfield.py

```diff
@@ -17,27 +17,24 @@
     JSONDecodeError = ValueError
 
 
-DEFAULT_JSON_SCHEMA = json.dumps({
-    'type': 'object',
-    'properties': {}
-})
+DEFAULT_JSON_SCHEMA = json.dumps({"type": "object", "properties": {}})
 
 
 class IJSONField(IField):
     """A text field that stores A JSON."""
 
-    schema = Attribute(
-        "schema",
-        _("The JSON schema string serialization.")
-    )
+    schema = Attribute("schema", _("The JSON schema string serialization."))
 
 
 @implementer(IJSONField, IFromUnicode)
 class JSONField(Field):
-
-    def __init__(self, schema=DEFAULT_JSON_SCHEMA, **kw):
+    def __init__(self, schema=DEFAULT_JSON_SCHEMA, widget=None, **kw):
         if not isinstance(schema, str):
             raise WrongType
+        if widget and not isinstance(widget, str):
+            raise WrongType
+
+        self.widget = widget
```

Comment:
Maybe go with "widget_name" ?

Reply:
mmm the serialisation already uses the "widget" naming (both in the adapters and in the property name)... :( being in the Python side, if no class is going to be specified there, ever... is it not too much specialisation?